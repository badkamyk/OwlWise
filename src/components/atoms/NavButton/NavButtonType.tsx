export type NavButtonType = {
	isExpanded: boolean;
	setIsExpanded: React.Dispatch<React.SetStateAction<boolean>>;
};
