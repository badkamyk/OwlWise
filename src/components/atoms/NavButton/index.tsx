import { NavButtonType } from "./NavButtonType";

export default function NavButton({
	isExpanded,
	setIsExpanded,
}: NavButtonType) {
	return (
		<button
			onClick={() => setIsExpanded(!isExpanded)}
			data-collapse-toggle="navbar-default"
			type="button"
			className="ml-3 inline-flex items-center rounded-lg p-2 text-sm text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600 md:hidden"
			aria-controls="navbar-default"
			aria-expanded={isExpanded}
		>
			<span className="sr-only">Open main menu</span>
			<svg
				className="h-6 w-6"
				aria-hidden="true"
				fill="currentColor"
				viewBox="0 0 20 20"
				xmlns="http://www.w3.org/2000/svg"
			>
				<path
					fillRule="evenodd"
					d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
					clipRule="evenodd"
				></path>
			</svg>
		</button>
	);
}
