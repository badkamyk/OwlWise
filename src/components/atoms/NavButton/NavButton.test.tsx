import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import NavButton from "./index";

it("NavButton with props", () => {
	const { container } = render(
		<NavButton isExpanded={true} setIsExpanded={() => {}} />
	);
	expect(
		container.getElementsByClassName(
			"ml-3 inline-flex items-center rounded-lg p-2 text-sm text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600 md:hidden"
		)
	).toBeTruthy();
	expect(container.getElementsByClassName("sr-only")).toBeTruthy();
	expect(screen.getByText("Open main menu")).toBeTruthy();
	expect(container).toMatchSnapshot();
});
