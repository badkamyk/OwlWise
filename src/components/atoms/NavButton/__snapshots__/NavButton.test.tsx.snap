// Vitest Snapshot v1

exports[`NavButton with props 1`] = `
<div>
  <button
    aria-controls="navbar-default"
    aria-expanded="true"
    class="ml-3 inline-flex items-center rounded-lg p-2 text-sm text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600 md:hidden"
    data-collapse-toggle="navbar-default"
    type="button"
  >
    <span
      class="sr-only"
    >
      Open main menu
    </span>
    <svg
      aria-hidden="true"
      class="h-6 w-6"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;
